src/data/mv_polynomial/cardinal.lean

```diff
@@ -89,14 +89,35 @@ calc #(mv_polynomial_fun σ R) = #R + #σ + #(ulift bool) :
 
 namespace mv_polynomial
 
+lemma cardinal_lift_mk_eq_max {σ : Type u} {R : Type v}
+  [nonempty σ] [comm_semiring R] [nontrivial R] :
+  #(mv_polynomial σ R) = max (max (cardinal.lift.{u} $ #R) $ cardinal.lift.{v} $ #σ) ℵ₀ :=
+begin
+  haveI : infinite (σ →₀ ℕ) := infinite_iff.2 ((le_max_right _ _).trans (mk_finsupp_nat σ).symm.le),
+  refine (mk_finsupp_lift_of_infinite _ R).trans _,
+  rw [mk_finsupp_nat, max_assoc, lift_max, lift_aleph_0, max_comm],
+end
+
+lemma cardinal_mk_eq_max {σ R : Type u} [nonempty σ] [comm_semiring R] [nontrivial R] :
+  #(mv_polynomial σ R) = max (max (#R) (#σ)) ℵ₀ :=
+by rw [cardinal_lift_mk_eq_max, lift_id, lift_id]
+
+lemma cardinal_lift_mk_le_max {σ : Type u} {R : Type v} [comm_semiring R] :
+  #(mv_polynomial σ R) ≤ max (max (cardinal.lift.{u} $ #R) $ cardinal.lift.{v} $ #σ) ℵ₀ :=
+begin
+  casesI subsingleton_or_nontrivial R,
+  { exact (le_one_iff_subsingleton.2 finsupp.coe_fn_injective.subsingleton).trans
```

Comment:
You're proving that multivariate polynomials over a subsingleton type are a subsingleton type themselves, right? This could be its own instance.

Reply:
Also, `finsupp.coe_fn_injective` is deprecated. Not sure what the fix is here.

Reply:
Is there a good reason this isn't an instance?

Also, I'm not the one who made the deprecation decision. That should be discussed on Zulip.

Reply:
mv_polynomial.unique is not an instance due to #6025, but polynomial.unique is actually an instance. Let's see if CI passes at https://github.com/leanprover-community/mathlib/compare/finsupp_unique?expand=1

Reply:
`unique` instances aren't within the scope of #6025, so they shouldn't be an issue. Also, the problem isn't really a build problem, but rather a performance one.

Reply:
The docstring of mv_polynomial.unique refers to gh-6025; I think it's because unique.subsingleton is an instance.

Reply:
I could open a PR from branch#finsupp_unique; it actually builds but now has a merge conflict. Would you prefer that be merged before this PR?

Reply:
Might be nice, yes.

Reply:
I've removed explicit invocation of `mv_polynomial.unique`. I also removed `mv_polynomial.cardinal_mk_eq_one` because it's now proved by simp (even before `cardinal_lift_mk_le_max` is proven), and for rewrites one can just use `cardinal.mk_eq_one`.